### PR TITLE
create and style Header

### PR DESCRIPTION
**Pull request by adrianixGit**

### CODE REVIEW

File: fischkapp/src/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { StyledHeader } from "./styles/Header.styled";
+export const Header = () => {
+  return (
+    <StyledHeader>
+      <div className="header__logo">
```
Comment (Sumick) [inline fischkapp/src/components/Header.tsx]: **nitpick:** Why styling by class names when using styled components? It's not a good practice - all styled elements should be created with styled-components. You will avoid such deep nesting

File: fischkapp/src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        <h1 className="header__logoName">FLASHCARDS</h1>
+        <p className="header__cardsNumber">0</p>
+      </div>
+      <button>
+        <p></p>
```
Comment (Sumick) [inline fischkapp/src/components/Header.tsx]: **nitpick:** We should use SVG icon here. You should be able to change your icon in button when needed. Now, I have no idea what icon is used in this button

File: fischkapp/src/components/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { StyledHeader } from "./styles/Header.styled";
+export const Header = () => {
+  return (
+    <StyledHeader>
+      <div className="header__logo">
+        <h1 className="header__logoName">FLASHCARDS</h1>
+        <p className="header__cardsNumber">0</p>
+      </div>
+      <button>
```
Comment (Sumick) [inline fischkapp/src/components/Header.tsx]: **nitpick:** Button should be also a styled component

File: fischkapp/src/components/styles/Header.styled.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  width: 100%;
+  height: 9vh;
+  position: fixed;
+  padding: 2.5vh;
+  box-shadow: 0px -16px 52px -10px rgba(152, 163, 207, 1);
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+
+  .header__logo {
```
Comment (Sumick) [inline fischkapp/src/components/styles/Header.styled.tsx]: **nitpick**: You should avoid such deep nesting of styles. Let's split it into smaller styled-components

File: fischkapp/src/components/styles/Header.styled.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    display: flex;
+    height: 100%;
+    align-items: center;
+
+    .header__logoName {
+    }
+
+    .header__cardsNumber {
```
Comment (Sumick) [inline fischkapp/src/components/styles/Header.styled.tsx]: **nitpick:** You don't need to use BEM naming when using styled-components. BEM is used for style scoping in big CSS files, but by using styled-components they are already scoped

File: fischkapp/src/components/styles/Header.styled.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      font-size: 25px;
+      margin-left: 0.5vw;
+      color: #98a3cf;
+    }
+  }
+
+  button {
```
Comment (Sumick) [inline fischkapp/src/components/styles/Header.styled.tsx]: **nitpick:** Button should be styled-component. Now you can't re-use it anywhere

**Review by Sumick — CHANGES_REQUESTED** [review]
Markdown looks good, but we need to improve `styled-components` architecture, because it currently looks messy, and non maintainable. You will find more details in comments below.

**Review by NerdbordMentor — APPROVED** [review]
Thank you for providing me with the code changes and task requirements. Based on the changes provided, I can see that the task goal was to create an app header with its contents, and the UI was to be implemented based on the design preview. 

The changes made to the `index.html` file added preconnect links to Google Fonts, which is a good practice for optimizing page speed. 

In the `App.tsx` file, the changes removed the default `Hello World` text and added the `Header` component wrapped in a `Wrapper` component. This is a good implementation of the header component as it is now reusable and can be used across multiple pages.

The `Header.tsx` file was added and contains the `Header` component implementation. The component includes the `StyledHeader`, `StyledLogo`, and `StyledAddButton` styled components. The `StyledHeader` component is a container for the `StyledLogo` and `StyledAddButton` components. The `StyledLogo` component contains the `h1` and `p` tags for the app name and card count, respectively. The `StyledAddButton` component is a button that contains an image of a plus sign. The implementation of the `Header` component is based on the design preview provided.

The `GlobalStyles.tsx`, `Header.styled.tsx`, and `Wrapper.tsx` files were added and contain the global styles, header styles, and wrapper styles, respectively. The global styles include a reset of all margins and paddings, box-sizing set to border-box, font-family set to Raleway, and letter-spacing set to 2px. The header styles include the styles for the `StyledHeader` and `StyledLogo` components, while the wrapper styles include the styles for the `Wrapper` component.

Overall, the changes made were well-implemented and followed best practices. The code is easy to read and understand, and the implementation of the `Header` component is based on the design preview provided.

**Pull request merged**